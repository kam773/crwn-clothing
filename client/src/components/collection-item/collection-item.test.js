import React from 'react';
import { shallow } from 'enzyme';
import { CollectionItem } from './collection-item.component';
import {
  AddButton,
  BackgroundImage,
  NameContainer,
  PriceContainer,
} from './collection-item.styles';
import renderer from 'react-test-renderer';

const mockProps = {
  addItem: jest.fn(),
  item: { name: 'name', price: '20', imageUrl: 'url' },
};

describe('create snapshot', () => {
  it('capture snapshot of CollectionItem', () => {
    const wrapper = renderer
      .create(<CollectionItem {...mockProps} />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('shallow render react component', () => {
  it('render the DUMB component', () => {
    const wrapper = shallow(<CollectionItem {...mockProps} />);
    expect(wrapper.length).toEqual(1);
  });
  it('render AddButton', () => {
    const wrapper = shallow(<CollectionItem {...mockProps} />);
    expect(wrapper.find(AddButton)).toBeDefined();
  });
  it('render BackgroundImage', () => {
    const wrapper = shallow(<CollectionItem {...mockProps} />);
    expect(wrapper.find(BackgroundImage)).toBeDefined();
  });
  it('render NameContainer', () => {
    const wrapper = shallow(<CollectionItem {...mockProps} />);
    expect(wrapper.find(NameContainer)).toBeDefined();
  });
  it('render PriceContainer', () => {
    const wrapper = shallow(<CollectionItem {...mockProps} />);
    expect(wrapper.find(PriceContainer)).toBeDefined();
  });
});
