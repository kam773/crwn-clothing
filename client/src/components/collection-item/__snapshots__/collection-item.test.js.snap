// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot capture snapshot of CollectionItem 1`] = `
<div
  className="sc-bwzfXH cFLGUT"
>
  <div
    className="sc-bxivhb dbDlmL image"
  />
  <div
    className="sc-ifAKCX kDnhkJ"
  >
    <span
      className="sc-EHOje kkQgTP"
    >
      name
    </span>
    <span
      className="sc-bZQynM gGKQi"
    >
      20
    </span>
  </div>
  <button
    className="sc-bdVaJa fKNtmD sc-htpNat iOWmNc"
    onClick={[Function]}
  >
    Add to cart
  </button>
</div>
`;
